invenio_rdm_records/services/communities/service.py

```diff
@@ -245,3 +262,94 @@ def search(
             extra_filter=communities_filter,
             **kwargs
         )
+
+    @staticmethod
+    def _get_excluded_communities_filter(record, identity, id_):
+        """Return filter to exclude communities that should not be suggested."""
+        communities_to_exclude = []
+        communities_ids = record.parent.communities.ids
+        for community_id in communities_ids:
+            communities_to_exclude.append(dsl.Q("term", **{"id": community_id}))
+
+        open_requests = current_requests_service.search(
+            identity,
+            extra_filter=dsl.query.Bool(
+                "must",
+                must=[
+                    dsl.Q("term", **{"topic.record": id_}),
+                    dsl.Q("term", **{"type": CommunityInclusion.type_id}),
+                    dsl.Q("term", **{"is_open": True}),
+                ],
+            ),
+        )
+
+        for request in open_requests.hits:
+            communities_to_exclude.append(
+                dsl.Q("term", **{"id": request["receiver"]["community"]})
+            )
+
+        return dsl.query.Bool("must_not", must_not=communities_to_exclude)
+
+    def search_suggested_communities(
+        self,
+        identity,
+        id_,
+        params=None,
+        search_preference=None,
+        expand=False,
+        extra_filter=None,
+        **kwargs
+    ):
+        """Search for communities that can be added to a record."""
+        record = self.record_cls.pid.resolve(id_)
+        self.require_permission(identity, "manage", record=record)
+
+        communities_filter = self._get_excluded_communities_filter(
+            record, identity, id_
+        )
+
+        if extra_filter is not None:
+            communities_filter = communities_filter & extra_filter
+
+        return current_communities.service.search(
+            identity,
+            params=params,
+            search_preference=search_preference,
+            expand=expand,
+            extra_filter=communities_filter,
+            **kwargs
+        )
+
+    def search_user_suggested_communities(
```

Comment:
I would not have added yet-another-service for this, and removed the UI filter `all/my` communities for this particular case.
In any case, if this stays, compared to the previous one it is simply adding a new filter.
All the searches here can put in common, and the filters parametrized.